pharus/component_interface.py

```diff
@@ -163,30 +167,36 @@ def __init__(self, *args, **kwargs):
     def dj_query_route(self):
         fetch_metadata = self.fetch_metadata
         record_header, table_records, total_count = _DJConnector._fetch_records(
-            query=fetch_metadata["query"] & self.restriction[0],
+            query=fetch_metadata["query"] & self.restriction,
             fetch_args=fetch_metadata["fetch_args"],
-            **{
-                k: (
-                    int(v)
-                    if k in ("limit", "page")
-                    else (
-                        v.split(",")
-                        if k == "order"
-                        else json.loads(b64decode(v.encode("utf-8")).decode("utf-8"))
-                    )
-                )
-                for k, v in request.args.items()
-            },
+            limit=int(request.args["limit"]) if "limit" in request.args else 1000,
+            page=int(request.args["page"]) if "page" in request.args else 1,
+            order=request.args["order"].split(",") if "order" in request.args else None,
         )
-        return dict(
-            recordHeader=record_header, records=table_records, totalCount=total_count
+
+        return NumpyEncoder.dumps(
+            dict(
+                recordHeader=record_header,
+                records=table_records,
+                totalCount=total_count,
+            )
         )
 
     def attributes_route(self):
         attributes_meta = _DJConnector._get_attributes(self.fetch_metadata["query"])
-        return dict(
-            attributeHeaders=attributes_meta["attribute_headers"],
-            attributes=attributes_meta["attributes"],
+        query = self.fetch_metadata["query"] & self.restriction[0]
+        unique_values = []
+        for key in query.heading.attributes.keys():
+            result = (dj.U(key) & query).fetch()
+            unique_values.append(
+                [dict({"text": str(x[0]), "value": x[0]}) for x in result]
+            )
+        return NumpyEncoder.dumps(
+            dict(
+                attributeHeaders=attributes_meta["attribute_headers"],
+                attributes=attributes_meta["attributes"],
+                unique_values=unique_values,
```

Comment:
Think this is best resolved in [_get_attributes](https://github.com/jverswijver/pharus/blob/6a427d2ab23c37962618e83d2fb58b424279b551/pharus/interface.py#L228).

See this [comment](url) for more context.